manifests/osfamily/solaris.pp

```diff
@@ -32,6 +32,57 @@
         source => "puppet:///pe_packages/${pe_server_version}/${::platform_tag}/solaris-noask",
       }
     }
+    '11': {
+      class { 'puppet_agent::prepare::package':
+        package_file_name => $package_file_name,
+      }
+      contain puppet_agent::prepare::package
+
+      # Backup user configuration because solaris 11 will blow away
+      # /etc/puppetlabs/ when uninstalling the pe-* modules.
+      file { '/tmp/puppet_agent/':
+        ensure => directory,
+      }
+      exec { 'puppet_agent backup /etc/puppetlabs/':
+        command => 'cp -r /etc/puppetlabs/ /tmp/puppet_agent/',
+        require => [Exec['puppet_agent copy packages'], File['/tmp/puppet_agent/']],
+        path    => '/bin:/usr/bin:/sbin:/usr/sbin',
+      }
+
+      $pkgrepo_dir = '/etc/puppetlabs/installer/solaris.repo'
+
+      exec { 'puppet_agent remove existing repo':
+        command   => "pkgrepo remove -s '${pkgrepo_dir}' '*'",
```

Comment:
@jpartlow So moving this to after the package uninstallation has occurred seems to not have an effect on anything. The whole of `/etc/puppetlabs/` is still removed making the backup and restoration of that directory still required for the puppet-agent package to install and for puppet to be in an okay state after the puppet run. We also still need to have the package dependencies in puppet code otherwise the run fails for attempting to uninstall the packages when other dependant packages are still installed.